examples/pytorch/diffusion/run_sd.py

```diff
@@ -23,6 +25,56 @@
 except:
     IPEX_AVAILABLE = False
 
+class SDDataset(Dataset):
+    def __init__(self, data_folder):
+        self._check_text_image_filename(data_folder)
+        self._images = ImageFolder(data_folder)
+        self._texts = self._read_text_files_in_folder(data_folder)
+
+    def transform(self, preprocess):
+        self._images = [preprocess(image.convert('RGB')) for image, _ in self._images]
+
+    def _check_text_image_filename(self, folder):
+        self._text_files = []
+        self._image_files = []
+        for file_name in os.listdir(folder):
+            if file_name.endswith('.txt'):
+                self._text_files.append(file_name)
+
+        image_folder = os.path.join(folder, 'jpg')
```

Comment:
@hongdami 
To use `ImageFolder`, creating `jpg` folder is very tricky.
i think we need to use [similar implementation](https://github.com/pytorch/vision/blob/main/torchvision/datasets/folder.py#L259) at `ImageFolder` to keep `laion400m` structure.
```
def pil_loader(path: str) -> Image.Image:
    # open path as file to avoid ResourceWarning (https://github.com/python-pillow/Pillow/issues/835)
    with open(path, "rb") as f:
        img = Image.open(f)
        return img.convert("RGB")
```